tests/resources/FunctionParameterTemplatingTest.nestml

```diff
@@ -0,0 +1,49 @@
+/**
+ *
+ *  FunctionParameterTemplatingTest.nestml
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  This test is used to test the correct derivation of types when functions use templated type parameters.
+*/
+
+
+neuron templated_function_parameters_type_test:
+    state:
+       force N
+       foo real
+    end
+
+    parameters:
+    end
+
+    update:
+        force = max(1 N, 10 N)
+        force = max(1 A, 10 N)
+        force = max(1 N, 10 A)
+        force = max(1 A, 10 A)
+        foo = max(1, 3.14)
```

Comment:
I'm unsure about where this is actually tested...
For me this file should generate:

- "fine" for the first and last lines
- "wrong" for all others but with different reasons